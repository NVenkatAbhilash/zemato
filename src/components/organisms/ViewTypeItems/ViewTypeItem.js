import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Grid } from "@material-ui/core";
import HeaderViewItem from "../../molecules/HeaderViewItem/HeaderViewItem";
import foodImg1 from "./../../../resources/images/collection2.jpg";
import foodImg2 from "./../../../resources/images/collection4.jpg";
import foodImg3 from "./../../../resources/images/collection3.jpg";
import foodImg4 from "./../../../resources/images/collection5.jpg";
import { Link } from "react-router-dom";

const useStyles = makeStyles({
  root: {
    padding: "40px 40px 80px 40px",
  },
});

const ViewTypeItem = (props) => {
  const classes = useStyles();

  return (
    <Grid container alignItems="stretch" justify="space-evenly" className={classes.root} spacing={2} {...props}>
      <Grid item md={3} sm={6} xs={12}>
        <Link to="/viewType/delivery" style={{ textDecoration: 'none' }}>
          <HeaderViewItem foodImg={foodImg1}>Order Food Online</HeaderViewItem>
        </Link>
      </Grid>
      <Grid item md={3} sm={6} xs={12}>
      <Link to="/viewType/dineOut" style={{ textDecoration: 'none' }}>
        <HeaderViewItem foodImg={foodImg2}>Go out for a meal</HeaderViewItem>
        </Link>
      </Grid>
      <Grid item md={3} sm={6} xs={12}>
      <Link to="/viewType/nightLife" style={{ textDecoration: 'none' }}>
        <HeaderViewItem foodImg={foodImg3}>
          Nightlife &#38; Clubs
        </HeaderViewItem></Link>
      </Grid>
      <Grid item md={3} sm={6} xs={12}>
        <HeaderViewItem foodImg={foodImg4}>Zomato Pro</HeaderViewItem>
      </Grid>
    </Grid>
  );
};

export default ViewTypeItem;
