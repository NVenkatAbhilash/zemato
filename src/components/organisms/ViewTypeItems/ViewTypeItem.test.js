import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import ViewTypeItem from "./ViewTypeItem";
import renderer from "react-test-renderer";

it("renders ViewTypeItem snapshot correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <ViewTypeItem />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders ViewTypeItem without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <ViewTypeItem />
    </BrowserRouter>,
    div
  );
});
