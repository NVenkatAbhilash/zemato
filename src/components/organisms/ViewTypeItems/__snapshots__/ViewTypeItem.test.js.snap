// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ViewTypeItem snapshot correctly 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-space-evenly"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/viewType/delivery"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-2 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <img
            alt="view type"
            className="makeStyles-image-3"
            src="collection2.jpg"
          />
        </div>
        <div
          className="MuiGrid-root makeStyles-content-4 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="text"
          >
            Order Food Online
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/viewType/dineOut"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-2 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <img
            alt="view type"
            className="makeStyles-image-3"
            src="collection4.jpg"
          />
        </div>
        <div
          className="MuiGrid-root makeStyles-content-4 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="text"
          >
            Go out for a meal
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <a
      href="/viewType/nightLife"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="MuiGrid-root makeStyles-root-2 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <img
            alt="view type"
            className="makeStyles-image-3"
            src="collection3.jpg"
          />
        </div>
        <div
          className="MuiGrid-root makeStyles-content-4 MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-h6"
            data-testid="text"
          >
            Nightlife & Clubs
          </h6>
        </div>
      </div>
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
  >
    <div
      className="MuiGrid-root makeStyles-root-2 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <img
          alt="view type"
          className="makeStyles-image-3"
          src="collection5.jpg"
        />
      </div>
      <div
        className="MuiGrid-root makeStyles-content-4 MuiGrid-item"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6"
          data-testid="text"
        >
          Zomato Pro
        </h6>
      </div>
    </div>
  </div>
</div>
`;
